test/Passes/tile-and-fuse_named-op.mlir

```diff
@@ -0,0 +1,131 @@
+// RUN: tpp-opt %s -split-input-file -tile-consumer-and-fuse-producers="tile-sizes=2,2 use-for-all=false" -cse | FileCheck %s
+
+// CHECK: func.func @matmul_sequence_fusion_expect_no_fusion
+func.func @matmul_sequence_fusion_expect_no_fusion(%arg0: tensor<32x64xf32>, %arg1: tensor<64x32xf32>,
+    %arg2: tensor<32x32xf32>, %arg3: tensor<32x64xf32>, %arg4: tensor<32x64xf32>,
+    %arg5: tensor<64x32xf32>, %arg6: tensor<32x32xf32>) -> tensor<32x32xf32> {
+  %0 = linalg.matmul ins(%arg0, %arg1 : tensor<32x64xf32>, tensor<64x32xf32>)
+    outs(%arg2 : tensor<32x32xf32>) -> tensor<32x32xf32> // [M, N0] * [N0, N1]
+  %1 = linalg.matmul ins(%0, %arg3 : tensor<32x32xf32>, tensor<32x64xf32>)
+    outs(%arg4 : tensor<32x64xf32>) -> tensor<32x64xf32> // [M, N1] * [N1, N2]
+  %2 = linalg.matmul ins(%1, %arg5 : tensor<32x64xf32>, tensor<64x32xf32>)
+    outs(%arg6 : tensor<32x32xf32>) -> tensor<32x32xf32> // [M, N2] * [N2, N3]
+  return %2 : tensor<32x32xf32>
+}
+
+// CHECK-COUNT-2: scf.for
+// CHECK: linalg.matmul
+// CHECK-COUNT-2: scf.for
+// CHECK: linalg.matmul
+// CHECK-COUNT-2: scf.for
+// CHECK: linalg.matmul
+
+// -----
+
+func.func @matmul_eletwise_matmul_and_relu(%arg0: tensor<32x64xf32>, %arg1: tensor<64x32xf32>,
+   %arg2: tensor<32x32xf32>) -> tensor<32x32xf32> {
+    %cst = arith.constant 0.000000e+00 : f32
+    %0 = linalg.matmul ins(%arg0, %arg1 : tensor<32x64xf32>, tensor<64x32xf32>) outs(%arg2 : tensor<32x32xf32>) -> tensor<32x32xf32>
+    %1 = tensor.empty() : tensor<32x32xf32>
+    %2 = linalg.fill ins(%cst : f32) outs(%1 : tensor<32x32xf32>) -> tensor<32x32xf32>
+    %3 = linalg.max ins(%0, %2 : tensor<32x32xf32>, tensor<32x32xf32>) outs(%arg2 : tensor<32x32xf32>) -> tensor<32x32xf32>
+    return %3 : tensor<32x32xf32>
+}
+
+// CHECK-DAG: #[[MAP:.+]] = affine_map<(d0, d1) -> (d0, d1)>
+// CHECK: func.func @matmul_eletwise_matmul_and_relu
+// CHECK-DAG: %[[C32:.+]] = arith.constant 32 : index
+// CHECK-DAG: %[[C0:.+]] = arith.constant 0 : index
+// CHECK-DAG: %[[C2:.+]] = arith.constant 2 : index
+// CHECK: %[[LOOP:.+]] = scf.for %{{.+}} = %[[C0]] to %[[C32]] step %[[C2]]
+// CHECK-NEXT: %[[LOOP1:.+]] = scf.for %{{.+}} = %[[C0]] to %[[C32]] step %[[C2]]
+// CHECK: linalg.matmul
+// CHECK-NEXT: tensor.empty()
+// CHECK-NEXT: linalg.fill
+// CHECK: linalg.generic
```

Comment:
Why is the matmul tiled as a named op and the max isn't?

Reply:
While doing the fusion in tile-and-fuse pass, at one point we are generalizing the NamedOps as "populateFoldUnitExtentDimsViaSlicesPatterns" works only with linalg.generic.
Hence linalg.max is getting generalized.

This is what I had pointed in our last discussion IIRC.

Reply:
Ok, this is good information that we'll need to fix upstream eventually. Can you add this as a TODO/FIXME somewhere in the code, so we don't forget? A Github issue would also work.

Reply:
Actually, this is our downstream pass "tile-consumer-and-fuse-producers".